packages/compass-query-history/src/stores/recent-list-store.js

```diff
@@ -53,12 +53,7 @@ const configureStore = (options = {}) => {
     },
 
     addRecent(recent) {
-      /* Ignore queries that don't have a namespace. */
-      if (!('ns' in recent)) {
-        return;
-      }
-
-      const ns = recent.ns;
+      const ns = this.state.ns;
```

Comment:
This plugin relied on ns being passed with `query-applied` event before even though this event is emitted on a local registry with the same namespace that is passed to the configureStore. Changed it to pick up the namespace from the plugin state instead of relying on the event information